public/app/features/plugins/sandbox/distortion_map.ts

```diff
@@ -0,0 +1,73 @@
+type DistortionMap = Map<unknown, unknown>;
+const generalDistortionMap: DistortionMap = new Map();
+
+function failToSet() {
+  throw new Error('Plugins are not allowed to set sandboxed properties');
+}
+
+// sets distortion to protect iframe elements
+function distordIframeAttributes(distortions: DistortionMap) {
+  const iframeHtmlForbiddenProperties = ['contentDocument', 'contentWindow', 'src', 'srcdoc', 'srcObject', 'srcset'];
+
+  for (const property of iframeHtmlForbiddenProperties) {
+    const descriptor = Object.getOwnPropertyDescriptor(HTMLIFrameElement.prototype, property);
+    if (descriptor) {
+      function fail() {
+        throw new Error('iframe.' + property + ' is not allowed in sandboxed plugins');
+      }
+      if (descriptor.value) {
+        distortions.set(descriptor.value, fail);
+      }
+      if (descriptor.set) {
+        distortions.set(descriptor.set, fail);
+      }
+      if (descriptor.get) {
+        distortions.set(descriptor.get, fail);
+      }
+    }
+  }
+}
+
+function distordConsole(distortions: DistortionMap) {
+  // distorts window.console to prefix it
+  const descriptor = Object.getOwnPropertyDescriptor(window, 'console');
+  if (descriptor?.value) {
+    function sandboxLog(...args: unknown[]) {
+      console.log(`[plugin]`, ...args);
```

Comment:
Maybe it would make sense to add the `pluginId` to the prefix as well to make identifying logs easier?

Reply:
It'll make things more complicated and less performant code-wise.

Distortion are cached and if we want to distort the console (or other methods) catered to specific pluginIds we'll have a performance penalty. That's why event though it makes sense I rather we keep a generic plugin prefix.

Reply:
I see, if it is a significant performance penalty then I think it makes sense. I guess this is something that we can eventually think about if we figure out that we would need it for debugging.